src/types/chain.ts

```diff
@@ -26,7 +36,7 @@ export interface ChainOperation {
  */
 export interface ChainMessage extends ChainOperation {
   /** Identifies this as a transaction operation */
-  operationType: 'transaction';
+  operationType: OperationType;
```

Comment:
If I fully specify `OperationType.TRANSACTION` here, then in the base implementations, the compiler complains and I'd need to cast

https://github.com/user-attachments/assets/b8556a63-2e7f-46b2-85bf-331a2863aa23